homeassistant/components/netatmo/sensor.py

```diff
@@ -145,6 +161,39 @@ def get_entities():
 
     async_add_entities(await hass.async_add_executor_job(get_entities), True)
 
+    @callback
+    def add_public_entities():
+        """Retrieve Netatmo public weather entities."""
+        entities = []
+        for area in entry.options.get(CONF_WEATHER_AREAS).values():
+            data = NetatmoPublicData(
+                auth,
+                lat_ne=area[CONF_LAT_NE],
+                lon_ne=area[CONF_LON_NE],
+                lat_sw=area[CONF_LAT_SW],
+                lon_sw=area[CONF_LON_SW],
+            )
+            for sensor_type in SUPPORTED_PUBLIC_SENSOR_TYPES:
+                entities.append(NetatmoPublicSensor(area, data, sensor_type,))
+
+        for device in async_entries_for_config_entry(device_registry, entry.entry_id):
+            if device.model == "Public Weather stations":
+                device_registry.async_remove_device(device.id)
+
+        if entities:
+            async_add_entities(entities)
+
+    async_dispatcher_connect(hass, "signal_update", add_public_entities)
+
+    entry.add_update_listener(async_config_entry_updated)
+
+    add_public_entities()
+
+
+async def async_config_entry_updated(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle signals of config entry being updated."""
+    async_dispatcher_send(hass, "signal_update")
```

Comment:
You don't really need to use dispatcher here, you could just retrieve your netatmo instance and call the add_public_entities method. See how I do it in deCONZ

Reply:
This will be worked over once https://github.com/home-assistant/core/pull/35571 is merged.